### ISSUE TITLE
Crashes due to older version of `fabric-loader` in Minecraft 1.20

### ISSUE THREAD

**Issue by itsKhalidHossain** [QuiltMC/quilt-loader#316]
Quilt Loader is incompatible with few mods in Minecraft 1.20. I'm using quilt loader 0.18.10. 

The crash report says,
```
Fabric API requires at least version 0.14.21 or any newer version of fabricloader, but only a different version is present: 0.14.19
```

AppleSkin, FerriteCore and Sound Physics Remastered are also incompatible from the mods that I have tried.

**Comment by ehrrsn7** [comment]
Basically all of the mods I'm using are throwing this error (which is very weird, because the QuiltMC client installer loads version 0.18.10)

**Comment by eristhea** [comment]
hey, this is because quilt loader has to provide a specific version of fabric loader for compatibility, which is currently `0.14.19`. these mods depend on newer versions of fabric loader that quilt doesn't provide yet. the good news is they probably don't *need* to do that, and you can solve the issue with [dependency overrides](https://github.com/QuiltMC/quilt-loader/wiki/Dependency-Overrides)!

**Comment by daFoggo** [comment]
can any one make a video how to fix this?

**Comment by LostLuma** [comment]
I don't think a video is really warranted, however the wiki page could definitely use an example for less experienced users.

You'll need to create the `config/quilt-loader-overrides.json` file as mentioned in the wiki, and then add the basic json structure into it. To then remove a required version that you do not think is necessary for running the game (such as Fabric Loader 0.14.21 in this case) add an object with a single key named "remove" and a nested dependency object in the `depends` array:

```json
{
  "schema_version": 1,
  "overrides": [
    {
      "path": "<mods>/ferritecore-6.0.0-fabric.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          }
        }
      ],
      "breaks": []
    }
  ]
}
```

Note that the `"versions": ">=0.14.21"` part is the same as the dependency in the `fabric.mod.json` / `quilt.mod.json` inside the mod's jar, if you are unsure what to use in the future. You can open jars with any program that can open zip files.

```json
  "depends": {
    "fabricloader": ">=0.14.21",
    "minecraft": ">=1.20 <1.21"
  },
```
<sub>Snippet from FerriteCore's `fabric.mod.json`, found inside the .jar file</sub>

```json
    "depends": [
      {
        "id": "quilt_loader",
        "versions": ">=0.18.10"
      },
```
<sub>Snippet from the [template mod's](https://github.com/QuiltMC/quilt-template-mod/blob/50b3c9e9b8559d1f237eea1e2dfe461afc2552dc/src/main/resources/quilt.mod.json#L24-L28) `quilt.mod.json`, found inside the .jar file</sub>

If multiple of your mods require a dependency override you can add more objects in the `overrides` array, separated by a comma each. The same goes for multiple dependencies on a single mod:

```json
{
  "schema_version": 1,
  "overrides": [
    {
      "path": "<mods>/ferritecore-6.0.0-fabric.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          }
        }
      ]
    },
    {
      "path": "<mods>/soundphysics-fabric-1.20-1.1.1.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          }
        },
        {
          "remove": {
            "id": "example_mod",
            "versions": ">=1.0.0"
          }
        }
      ]
    }
  ]
}
```

In place of the "remove" action you can also use the other two override types mentioned on the wiki, this might be useful if you're sending your override file to another user or including it in a modpack, so users can not install incompatible mods that you've simply removed.

```json
{
  "schema_version": 1,
  "overrides": [
    {
      "path": "<mods>/example-1.0.0.jar",
      "depends": [
        {
          "add": {
            "id": "example_one",
            "versions": ">=2.0.5"
          }
        },
        {
          "replace": {
            "id": "example_two",
            "versions": ">=1.5.0"
          },
          "with": {
            "id": "example_two",
            "versions": ">=1.4.0"
          }
        }
      ],
      "breaks": [
        {
          "remove": {
            "id": "example_three",
            "versions": ">=2.6.5"
          }
        }
      ]
    }
  ]
}
```

**Comment by rezelyn** [comment]
> I don't think a video is really warranted, however the wiki page could definitely use an example for less experienced users.
> 
> You'll need to create the `config/quilt-loader-overrides.json` file as mentioned in the wiki, and then add the basic json structure into it. To then remove a required version that you do not think is necessary for running the game (such as Fabric Loader 0.14.21 in this case) add an object with a single key named "remove" and a nested dependency object in the `depends` array:
> 
> ```json
> {
>   "schema_version": 1,
>   "overrides": [
>     {
>       "path": "<mods>/ferritecore-6.0.0-fabric.jar",
>       "depends": [
>         {
>           "remove": {
>             "id": "fabricloader",
>             "versions": ">=0.14.21"
>           }
>         }
>       ],
>       "breaks": []
>     }
>   ]
> }
> ```
> 
> Note that the `"versions": ">=0.14.21"` part is the same as the dependency in the `fabric.mod.json` / `quilt.mod.json` inside the mod's jar, if you are unsure what to use in the future. You can open jars with any program that can open zip files.
> 
> ```json
>   "depends": {
>     "fabricloader": ">=0.14.21",
>     "minecraft": ">=1.20 <1.21"
>   },
> ```
> 
> Snippet from FerriteCore's `fabric.mod.json`, found inside the .jar file
> 
> ```json
>     "depends": [
>       {
>         "id": "quilt_loader",
>         "versions": ">=0.18.10"
>       },
> ```
> 
> Snippet from the [template mod's](https://github.com/QuiltMC/quilt-template-mod/blob/50b3c9e9b8559d1f237eea1e2dfe461afc2552dc/src/main/resources/quilt.mod.json#L24-L28) `quilt.mod.json`, found inside the .jar file
> 
> If multiple of your mods require a dependency override you can add more objects in the `overrides` array, separated by a comma each. The same goes for multiple dependencies on a single mod:
> 
> ```json
> {
>   "schema_version": 1,
>   "overrides": [
>     {
>       "path": "<mods>/ferritecore-6.0.0-fabric.jar",
>       "depends": [
>         {
>           "remove": {
>             "id": "fabricloader",
>             "versions": ">=0.14.21"
>           }
>         }
>       ]
>     },
>     {
>       "path": "<mods>/soundphysics-fabric-1.20-1.1.1.jar",
>       "depends": [
>         {
>           "remove": {
>             "id": "fabricloader",
>             "versions": ">=0.14.21"
>           }
>         },
>         {
>           "remove": {
>             "id": "example_mod",
>             "versions": ">=1.0.0"
>           }
>         }
>       ]
>     }
>   ]
> }
> ```
> 
> In place of the "remove" action you can also use the other two override types mentioned on the wiki, this might be useful if you're sending your override file to another user or including it in a modpack, so users can not install incompatible mods that you've simply removed.
> 
> ```json
> {
>   "schema_version": 1,
>   "overrides": [
>     {
>       "path": "<mods>/example-1.0.0.jar",
>       "depends": [
>         {
>           "add": {
>             "id": "example_one",
>             "versions": ">=2.0.5"
>           }
>         },
>         {
>           "replace": {
>             "id": "example_two",
>             "versions": ">=1.5.0"
>           },
>           "with": {
>             "id": "example_two",
>             "versions": ">=1.4.0"
>           }
>         }
>       ],
>       "breaks": [
>         {
>           "remove": {
>             "id": "example_three",
>             "versions": ">=2.6.5"
>           }
>         }
>       ]
>     }
>   ]
> }
> ```

I got the same problem with a lot of mods (AppleSkin, Head Name Fix, Chunky, Open Parties and Claims...)
Works fine now, thanks for the help!

**Comment by daFoggo** [comment]
thank you so much man. But I have one last stupid question, where do I have to put this file in? I tried putting it directly in "D:\Minecraft\game" but i didn't change anything

**Comment by LostLuma** [comment]
It goes in the `config` directory inside the game directory, which by default is `.minecraft` in your `%AppData%` folder.
For context on my system the full path using the vanilla launcher would be `C:\Users\lilly\AppData\Roaming\.minecraft\config`.

What you linked may be where the launcher itself is located on your system, but I'm not sure as I use a third-party launcher.

**Comment by ziaobank** [comment]
Still not working. Can u check the code for me? 

{
  "schema_version": 1,
  "overrides": [
    {
      "path": "<mods>/ferritecore-6.0.0-fabric.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          }
        }
      ]
    },
    {
      "path": "<mods>/language-reload-1.5.7+1.20.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          } 
        }
      ]
    },
    {
      "path": "<mods>/simple-hud-enhanced+1.20-1.1.0.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"        
          }
        }      
      ]
    }
  ]
}

**Comment by daFoggo** [comment]
Still not working. Can u check the code for me?

{
  "schema_version": 1,
  "overrides": [
    {
      "path": "<mods>/ferritecore-6.0.0-fabric.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          }
        }
      ]
    },
    {
      "path": "<mods>/language-reload-1.5.7+1.20.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"
          } 
        }
      ]
    },
    {
      "path": "<mods>/simple-hud-enhanced+1.20-1.1.0.jar",
      "depends": [
        {
          "remove": {
            "id": "fabricloader",
            "versions": ">=0.14.21"        
          }
        }      
      ]
    }
  ]
}

**Comment by LostLuma** [comment]
You need to keep the `<mods>/` part of the path, it refers to the `mods` directory you put your mods in.

**Comment by AlexIIL** [comment]
This should be fixed in `0.19.0-beta.19`

**Comment by itsKhalidHossain** [comment]
> This should be fixed in `0.19.0-beta.19`

Thank you.
